### PR TITLE
10116 health check stability

### PR DESCRIPTION
**Pull request by TomElliottFlexion**
Adds a chron function which is constantly pinging our AWS services in both east/west and storing the applications health in dynamo.

Separately we have a health check which is fetching our applications health from dynamo every 30s to see if we should failover to a different region.

### CODE REVIEW

**Review by zachrog — APPROVED** [review]

File: shared/src/business/useCases/health/getCachedHealthCheckInteractor.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { StoredApplicationHealth } from '../../../persistence/dynamo/deployTable/setStoredApplicationHealth';
+
+export const getCachedHealthCheckInteractor = async (
+  applicationContext: IApplicationContext,
+): Promise<StoredApplicationHealth> => {
+  console.log('getCachedHealthCheckInteractor 1: ', process.env.REGION);
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/health/getCachedHealthCheckInteractor.ts]: Nit: Console.logs

File: shared/src/business/useCases/health/setHealthCheckCacheInteractor.test.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    applicationContext
+      .getUseCases()
+      .getHealthCheckInteractor.mockResolvedValue(applicationHealth);
+  });
+
+  afterAll(() => {
+    process.env.REGION = oldRegion;
+  });
+
+  it('should get service statuses and write an healthy status to persistence when all services are health', async () => {
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/health/setHealthCheckCacheInteractor.test.ts]: `when all services are healthy`?

Comment (TomElliottFlexion) [reply shared/src/business/useCases/health/setHealthCheckCacheInteractor.test.ts]: Good catch! Thank you

File: shared/src/business/useCases/health/getCachedHealthCheckInteractor.ts (hunk 1 above)
Comment (TomElliottFlexion) [reply shared/src/business/useCases/health/getCachedHealthCheckInteractor.ts]: Agreed, don't need these in test

File: web-api/terraform/api/cron.tf [hunk 3]
```diff
@@ -1,5 +1,6 @@
 resource "aws_lambda_function" "check_case_cron_lambda" {
-  count            = var.create_cron
+  # Replace with create_check_case_cron?
```
Comment (rachelschneiderman) [inline web-api/terraform/api/cron.tf]: Looks like this has been done! 🎉 Can the comment be removed?

Comment (TomElliottFlexion) [reply web-api/terraform/api/cron.tf]: TY

**Pull request merged**